src/mito2/src/sst/parquet.rs

```diff
@@ -453,30 +456,68 @@ mod tests {
         let writer_props = props_builder.build();
 
         let write_format = WriteFormat::new(metadata.clone());
-        let string = file_path.clone();
+        let fields: Vec<_> = write_format
+            .arrow_schema()
+            .fields()
+            .into_iter()
+            .map(|field| {
+                let data_type = field.data_type().clone();
+                if data_type == DataType::Binary {
+                    Field::new(field.name(), DataType::LargeBinary, field.is_nullable())
+                } else {
+                    Field::new(field.name(), data_type, field.is_nullable())
+                }
+            })
+            .collect();
+
+        let arrow_schema = Arc::new(Schema::new(fields));
+
+        // Ensures field_0 has LargeBinary type.
+        assert_eq!(
+            DataType::LargeBinary,
+            arrow_schema
+                .field_with_name("field_0")
+                .unwrap()
+                .data_type()
+                .clone()
+        );
         let mut buffered_writer = BufferedWriter::try_new(
-            string,
+            file_path.clone(),
             object_store.clone(),
-            write_format.arrow_schema(),
+            arrow_schema.clone(),
             Some(writer_props),
             write_opts.write_buffer_size.as_bytes() as usize,
             DEFAULT_WRITE_CONCURRENCY,
         )
         .await
         .unwrap();
-        let batch = new_batch_with_binary_by_range(&["a"], 0, 60);
-        let arrow_batch = write_format.convert_batch(&batch).unwrap();
-
-        buffered_writer.write(&arrow_batch).await.unwrap();
 
+        let batch = new_batch_with_large_binary(&["a"], 0, 60);
+        let arrow_batch = write_format.convert_batch(&batch).unwrap();
+        let arrays: Vec<_> = arrow_batch
+            .columns()
+            .iter()
+            .map(|array| {
+                let data_type = array.data_type().clone();
+                if data_type == DataType::Binary {
+                    arrow::compute::cast(array, &DataType::LargeBinary).unwrap()
+                } else {
+                    array.clone()
+                }
+            })
+            .collect();
+        let result = RecordBatch::try_new(arrow_schema, arrays).unwrap();
+
+        buffered_writer.write(&result).await.unwrap();
         buffered_writer.close().await.unwrap();
+
         let builder = ParquetReaderBuilder::new(FILE_DIR.to_string(), handle.clone(), object_store);
         let mut reader = builder.build().await.unwrap();
         check_reader_result(
             &mut reader,
             &[
-                new_batch_with_binary_by_range(&["a"], 0, 50),
-                new_batch_with_binary_by_range(&["a"], 50, 60),
+                new_batch_with_large_binary(&["a"], 0, 50),
+                new_batch_with_large_binary(&["a"], 50, 60),
```

Comment:
Sorry, I made a mistake in the previous suggestion. We should rename it to `new_batch_with_binary`
```suggestion
                new_batch_with_binary(&["a"], 0, 50),
                new_batch_with_binary(&["a"], 50, 60),
```